cmake/TuttleMacros.cmake

```diff
@@ -111,20 +111,40 @@ function(tuttle_ofx_plugin_target PLUGIN_NAME)
         # Plugin target is a shared library
         add_library(${PLUGIN_NAME} MODULE ${PLUGIN_SOURCES})
 
+        # Get plugin type
+        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} "image/io" IS_IOPLUGIN)
+        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} "image/display" IS_DISPLAY_PLUGIN)
+        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} "image/generator" IS_GENERATOR_PLUGIN)
+        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} "image/process" IS_PROCESS_PLUGIN)
+        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} "param/analysis" IS_ANALYSIS_PLUGIN)
+        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} "param/debug" IS_DEBUG_PLUGIN)
+
+        # Add this new plugin to custom Makefile target
+        add_dependencies(ofxPlugins ${PLUGIN_NAME})
+        if(IS_IOPLUGIN GREATER -1)
+            add_dependencies(ofxIO ${PLUGIN_NAME})
```

Comment:
Maybe if I add __TuttleOFX/plugins/__ before each path to check the type of the plugin? With this update, no way to compile an external plugin and add it with one of these new targets.